flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultScheduler.java

```diff
@@ -299,6 +302,21 @@ private void handleTaskFailure(
         final FailureHandlingResult failureHandlingResult =
                 executionFailureHandler.getFailureHandlingResult(
                         executionVertexId, error, timestamp);
+
+        // Notify shuffle master that the cached intermediate dataset is corrupted.
+        if (failureHandlingResult.getError() instanceof CacheCorruptedException) {
```

Comment:
I think it might be not sufficient to only consider `DefaultScheduler`. We might also have AdaptiveBatchScheduler for batch jobs and AdaptiveScheduler for streaming jobs (if we also consider the case to start a new streaming job with cached result partition).